test/e2e/main.grpc.test.ts

```diff
@@ -267,6 +270,93 @@ describe('grpc/main', () => {
+      // expect(conf.items.filter(i => i.key == "myconfigkey1")[0].metadata).toHaveProperty("hello");
+    });
+
+    // // @todo: figure out, subscribe doesn't pass keys but doesn't listen to anything?
+    // // https://github.com/dapr/dapr/issues/4529
+    // it('should be able to subscribe to configuration item changes on all keys', async () => {
+    //   const m = jest.fn(async (res: SubscribeConfigurationResponse) => { });
+
+    //   console.log("Creating Subscription");
+    //   await client.configuration.subscribe("config-redis", m);
+
+    //   // Change an item
+    //   await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey3 mynewvalue||1");
+
+    //   expect(m.mock.calls.length).toEqual(1);
+    // });
+
+    it('should be able to subscribe to configuration item changes on specific keys', async () => {
+      const m = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1", "myconfigkey2"], m);
+      await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey1 key1_mynewvalue||1");
+
+      expect(m.mock.calls.length).toEqual(1);
+      expect(m.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m.mock.calls[0][0].items[0].value).toEqual("key1_mynewvalue");
+    });
+
+    it('should be able to subscribe to configuration items through multiple streams', async () => {
+      const m1 = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+      const m2 = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1"], m1);
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1"], m2);
+
+      await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey1 key1_mynewvalue||1");
+
+      expect(m1.mock.calls.length).toEqual(1);
+      expect(m1.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m1.mock.calls[0][0].items[0].value).toEqual("key1_mynewvalue");
+
+      expect(m2.mock.calls.length).toEqual(1);
+      expect(m2.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m2.mock.calls[0][0].items[0].value).toEqual("key1_mynewvalue");
+    });
+
+    it('should be able to subscribe to configuration item changes on specific keys through a second stream', async () => {
+      const m = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1"], m);
+      await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey1 key1_mynewvalue||1");
+
+      expect(m.mock.calls.length).toEqual(1);
+      expect(m.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m.mock.calls[0][0].items[0].value).toEqual("key1_mynewvalue");
+    });
+  });
+
```

Comment:
Can we add a test for `subscribeWithMetadata`?

Reply:
Yup adding test for subscribeWithMetadata should be good.